### DISCUSSION TITLE
Displaying sidebar values in human readable format in PsiStudio

### DISCUSSION THREAD

**Discussion by xiangzhi** [microsoft/psi#150] [Ideas]
Currently, all values shown in the sidebar are always expressed in their base units (bytes/seconds). One of the difficulty with it is that when it is 9-10 digits long, it gets hard to parse (85432353).

Is there any plan to have an option to make it more readable in the future? Example: (85432353 -> 85.4324 MB)

I assume it is complicated because it has to be a global option with the ability to be toggled locally for each value. I'm curious what you all think about this! Maybe PsiStudio needs a setting page?

**Comment by danbohus** [comment]
I assume you're referring to the various stream properties displayed on the right hand __Properties__ tab, right? If so, yes, I think you're right. They are difficult to read, and it would be desirable at different times (and for different properties) to have them displayed differently. 

For now, we went with the idea that the auxiliary stream/partition/session/dataset statistics displayed via right click on items in the __Dataset__ tab are shown in abbreviated form for ease of use (and we can expand that set with other properties), while the properties on the right display full accuracy numbers. However, I can see how in some cases you might want some of those to display in a more readable form. 

And yes, something like that does belongs in settings. PsiStudio already has a settings file so a starting point (even in the absence of a setting dialog box) might be to expand that with additional settings for *property display format*s, perhaps indexed by property name somehow. 

As for the settings page, yes, some of the properties already existing in the settings (not necessarily all, e.g. the window position) should also be surfaced and editable from a settings dialog box, which we don't have yet. We could probably start with something simple for this dialog box, where it's just a grid of property-name (label) -> property-value (edit box) for the settings, and then figure out a way to organize the "names" of the various *property display formats*, e.g. `PropertyDisplayFormat.StreamSize` = `summary`, and so forth... Just some initial ideas, I'm sure this could be refined. 

As an aside, there is a [SizeFormatHelper](https://github.com/microsoft/psi/blob/master/Sources/Visualization/Microsoft.Psi.Visualization.Windows/Helpers/SizeFormatHelper.cs) class in the codebase that provides a helper method for formatting sizes, which is being used when showing stream properties in the __Datasets__ tab. In our next release we plan to include helpers that format latency and throughput numbers as well (and we could push that to the repo directly sooner if needed)

We don't have immediate plans to develop this _property display format_ or settings page features, but they represent a good community contribution opportunity. If someone starts working on this with the goal of creating a pull request, please mention this by replying below and creating a corresponding issue, and we're happy to discuss more the design of the feature in this thread.
